src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/SkipExecution.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement
+
+import com.amazon.opendistroforelasticsearch.indexmanagement.util.OpenForTesting
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.action.ActionListener
+import org.elasticsearch.action.admin.cluster.node.info.NodesInfoAction
+import org.elasticsearch.action.admin.cluster.node.info.NodesInfoRequest
+import org.elasticsearch.action.admin.cluster.node.info.NodesInfoResponse
+import org.elasticsearch.action.admin.cluster.node.info.PluginsAndModules
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.ClusterChangedEvent
+import org.elasticsearch.cluster.ClusterStateListener
+import org.elasticsearch.cluster.service.ClusterService
+
+@OpenForTesting
+class SkipExecution(
+    private val client: Client,
+    private val clusterService: ClusterService
+) : ClusterStateListener {
+    private val logger = LogManager.getLogger(javaClass)
+
+    @Volatile final var flag: Boolean = false
+        private set
+
+    init {
+        clusterService.addListener(this)
+    }
+
+    override fun clusterChanged(event: ClusterChangedEvent) {
+        if (event.nodesChanged() || event.isNewCluster) {
+            sweepISMPluginVersion()
+        }
+    }
+
+    fun sweepISMPluginVersion() {
+        // if old version ISM plugin exists (2 versions ISM in one cluster), set skip flag to true
+        val request = NodesInfoRequest().clear().addMetric("plugins")
+        client.execute(NodesInfoAction.INSTANCE, request, object : ActionListener<NodesInfoResponse> {
+            override fun onResponse(response: NodesInfoResponse) {
+                flag = false
```

Comment:
Probably should not reset flag at start. What happens when:
Cluster has multiple versions when new node is added and this logic is executed and sets flag to true.
As job is running another node is added and this sets flag back to false while checking and job is running and goes through. Small race condition, but could happen.

Reply:
that's some good thinking! I move it to 
```
if (versionSet.size > 1) {
    flag = true
} else flag = false
```